tools/src/main/java/org/neo4j/tools/txlog/CheckTxLogs.java

```diff
@@ -0,0 +1,167 @@
+            try ( LogEntryCursor logEntryCursor = LogTestUtils.openLog( fs, log ) )
+            {
+                while ( logEntryCursor.next() )
+                {
+                    LogEntry entry = logEntryCursor.get();
+                    if ( entry instanceof LogEntryCommand )
+                    {
+                        Command command = ((LogEntryCommand) entry).getXaCommand();
+                        if ( check.commandClass().isInstance( command ) )
+                        {
+                            long logVersion = PhysicalLogFiles.getLogVersion( log );
+                            process( check.commandClass().cast( command ), check, state, logVersion );
+                        }
+                    }
+                    commandsRead++;
+                }
+            }
+            out.println( "Processed " + log.getCanonicalPath() + " with " + commandsRead + " commands" );
+            out.println( state );
+        }
+    }
+
+    private <C extends Command, R extends Abstract64BitRecord> void process( C command, CheckType<C,R> check,
+            ProcessedRecords<R> state, long logVersion )
+    {
+        R before = check.before( command );
+        R after = check.after( command );
+
+        if ( !state.isValid( before ) )
+        {
+            LogRecord<R> seen = state.get( before.getId() );
+            LogRecord<R> current = new LogRecord<>( before, logVersion );
+
+            inconsistenciesHandler.handle( seen, current );
+        }
+
+        state.put( after, logVersion );
+    }
+
+    private static File parseDir( String[] args, PrintStream out )
+    {
+        if ( args.length != 1 )
+        {
+            printUsageAndExit( out );
+        }
+        File dir = new File( args[0] );
+        if ( !dir.isDirectory() )
+        {
+            out.println( "Invalid directory: '" + dir + "'" );
+            printUsageAndExit( out );
+        }
+        return dir;
+    }
+
+    private static File[] txLogsIn( File dir )
+    {
+        File[] logs = dir.listFiles( LogFiles.FILENAME_FILTER );
+        Arrays.sort( logs, new Comparator<File>()
+        {
+            @Override
+            public int compare( File f1, File f2 )
+            {
+                long f1Version = PhysicalLogFiles.getLogVersion( f1 );
+                long f2Version = PhysicalLogFiles.getLogVersion( f2 );
+                return Long.compare( f1Version, f2Version );
+            }
+        } );
+        return logs;
+    }
+
+    private static void printUsageAndExit( PrintStream out )
+    {
+        out.println( "Tool expects single argument - directory with tx logs" );
+        System.exit( 1 );
+    }
+
+    private static InconsistenciesHandler throwIfInconsistent()
+    {
+        return new InconsistenciesHandler()
+        {
+            @Override
+            public void handle( LogRecord<?> seen, LogRecord<?> current )
+            {
+                throw new IllegalStateException(
```

Comment:
In theory there might be lots of inconsistencies. What about just printing them by default?